aws-rds-optiongroup/src/main/java/software/amazon/rds/optiongroup/ReadHandler.java

```diff
@@ -45,19 +51,39 @@ protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
                 ))
                 .done((describeRequest, describeResponse, proxyInvocation, model, context) -> {
                     final OptionGroup optionGroup = describeResponse.optionGroupsList().stream().findFirst().get();
-                    final List<OptionConfiguration> optionConfigurations = Translator.translateOptionConfigurationsFromSdk(optionGroup.options());
+                    final List<Option> overriddenConfigurations = getOverriddenOptionConfigurations(optionGroup);
+
                     final List<Tag> tags = listTags(proxyInvocation, optionGroup.optionGroupArn());
                     return ProgressEvent.success(
                             ResourceModel.builder()
                                     .optionGroupName(optionGroup.optionGroupName())
                                     .engineName(optionGroup.engineName())
                                     .majorEngineVersion(optionGroup.majorEngineVersion())
                                     .optionGroupDescription(optionGroup.optionGroupDescription())
-                                    .optionConfigurations(optionConfigurations)
+                                    .optionConfigurations(Translator.translateOptionConfigurationsFromSdk(overriddenConfigurations))
                                     .tags(tags)
                                     .build(),
                             context
                     );
                 });
     }
+
+    private List<Option> getOverriddenOptionConfigurations(final OptionGroup optionGroup) {
+        final List<Option> overriddenOptions = new ArrayList<>();
+        if (CollectionUtils.isNotEmpty(optionGroup.options())) {
+            for (final Option option : optionGroup.options()) {
+                final List<OptionSetting> optionSettings = new ArrayList<>();
+                if (CollectionUtils.isNotEmpty(option.optionSettings())) {
+                    for (final OptionSetting optionSetting : option.optionSettings()) {
+                        final String defaultValue = optionSetting.defaultValue();
```

Comment:
what will happen if option is listed in Customer's stack definition, but value is matching the default setting?

Reply:
If customer got drift in this case, I think that will be a bug. I understand that we can not check stack Options as read handler input will contains only resource identifier without stack attributes. We might need CFN team assistance to provide a good way to handle default attributes.